pkg/plugins/virtual/virtual_plugin.go

```diff
@@ -0,0 +1,141 @@
+package main
+
+import (
+	"os/exec"
+	"reflect"
+	"syscall"
+
+	"github.com/golang/glog"
+	sriovnetworkv1 "github.com/openshift/sriov-network-operator/api/v1"
+	"github.com/openshift/sriov-network-operator/pkg/utils"
+)
+
+// VirtualPlugin Plugin type to use on a virtual platform
+type VirtualPlugin struct {
+	PluginName     string
+	SpecVersion    string
+	DesireState    *sriovnetworkv1.SriovNetworkNodeState
+	LastState      *sriovnetworkv1.SriovNetworkNodeState
+	LoadVfioDriver uint
+}
+
+const (
+	unloaded = iota
+	loading
+	loaded
+)
+
+// Plugin VirtualPlugin type
+var Plugin VirtualPlugin
+
+// Initialize our plugin and set up initial values
+func init() {
+	Plugin = VirtualPlugin{
+		PluginName:     "virtual_plugin",
+		SpecVersion:    "1.0",
+		LoadVfioDriver: unloaded,
+	}
+}
+
+// Name returns the name of the plugin
+func (p *VirtualPlugin) Name() string {
+	return p.PluginName
+}
+
+// Spec returns the version of the spec expected by the plugin
+func (p *VirtualPlugin) Spec() string {
+	return p.SpecVersion
+}
+
+// OnNodeStateAdd Invoked when SriovNetworkNodeState CR is created, return if need dain and/or reboot node
+func (p *VirtualPlugin) OnNodeStateAdd(state *sriovnetworkv1.SriovNetworkNodeState) (needDrain bool, needReboot bool, err error) {
+	glog.Info("virtual-plugin OnNodeStateAdd()")
+	needDrain = false
+	needReboot = false
+	err = nil
+	p.DesireState = state
+
+	needDrain = needDrainNode(state.Spec.Interfaces, state.Status.Interfaces)
+
+	if p.LoadVfioDriver != loaded {
+		if needVfioDriver(state) {
+			p.LoadVfioDriver = loading
+		}
+	}
+	return
+}
+
+// OnNodeStateChange Invoked when SriovNetworkNodeState CR is updated, return if need dain and/or reboot node
+func (p *VirtualPlugin) OnNodeStateChange(old, new *sriovnetworkv1.SriovNetworkNodeState) (needDrain bool, needReboot bool, err error) {
+	glog.Info("virtual-plugin OnNodeStateChange()")
+	needDrain = false
+	needReboot = false
+	err = nil
+	p.DesireState = new
+
+	if p.LoadVfioDriver != loaded {
+		if needVfioDriver(new) {
+			p.LoadVfioDriver = loading
```

Comment:
@atyronesmith I noticed the iommu kernel args change was removed, is it handled by openstack ?